### PR TITLE
feat: add S3 bucket CORS policy and bucket policy

### PR DESCRIPTION
**Pull request by ifranch**
Addresses https://github.com/metrotranscom/doorway/issues/49.

Add CORS policy to the S3 user uploads bucket that allows cross origin requests from the partners site URL for GET and PUT.
Add bucket policy that allows public access for GetObject and PutObject for objects in the user uploads bucket.

I was not able to run terraform plan to verify the changes because it seems like I don't have the required permissions for some existing configuration:

Error: reading IAM Role (AWSServiceRoleForRDS): AccessDenied: User: arn:aws:iam::364076391763:user/inesfranch@google.com is not authorized to perform: iam:GetRole on resource: role AWSServiceRoleForRDS because no identity-based policy allows the iam:GetRole action

### CODE REVIEW

File: bloom-instance/s3.tf
Comment (chriscasto) [inline bloom-instance/s3.tf:11]: This should not be a hardcoded value.  It would not, for example, work with the services currently running in our AWS environment.  It should be enough to just add a var for this of type `list(string)`.

File: bloom-instance/s3.tf
Comment (chriscasto) [inline bloom-instance/s3.tf:10]: Why do we need to allow cross-origin PUTs?  Aren't all PUTs done on the server side?

File: bloom-instance/s3.tf
Comment (chriscasto) [inline bloom-instance/s3.tf:32]: We definitely do not want unauthenticated users to upload files directly to S3.  All PUTs should be fully authenticated, even to the public bucket.

**Review by chriscasto — CHANGES_REQUESTED** [review]

**Comment by chriscasto** [comment]
I have a branch that I worked on over a month ago to create separate buckets for public and private files that I completely forgot about.  I just pushed it and created a PR here: #93 

If you want to wait until that branch gets merged into main, it would address at least one item above

File: bloom-instance/s3.tf
Comment (ifranch) [reply bloom-instance/s3.tf:10]: We do need to allow cross-origin PUTs. Without this change, PUT requests fail with this error:

Access to XMLHttpRequest at '[...]&x-id=PutObject' from origin 'http://localhost:3001' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Comment (chriscasto) [reply bloom-instance/s3.tf:10]: This warrants additional investigation.  I can't imagine a scenario where there would be a localhost origin set on a request originating from server-side code.  Are you sure there isn't something making a PUT from the client side?  Because that's all kinds of bad news even for the current Bloom instance if it is

**Comment by isaac-looker** [comment]
Are we able to close this? I was under the impression we're going to not upload files from the client.

**Pull request closed without merging**
